### PR TITLE
perf: Clean up root trajectory writer

### PR DESCRIPTION
**Pull request by XiaocongAi**
This PR has the following changes:

- Adds the writing of corresponding truth particle info of a track in the `RootTrajectorySummaryWriter`
- Adds the `RootTrajectorySummaryReader`
- With the above changes, the `RootTrajectorySummaryXXX` supercede the `RootTrajectoryParametersXXX`. Hence, the `RootTrajectoryParametersXXX` are removed.
- Update of the examples to use the `RootTrajectorySummaryXXX`

### CODE REVIEW

**Comment by XiaocongAi** [comment]
Hi @baschlag , do you have time to have a look?

**Comment by baschlag** [comment]
> Hi @baschlag , do you have time to have a look?

Sure I will have a look!

**Comment by baschlag** [comment]
Hi @XiaocongAi, I only have some minor comments. Otherwise it looks uncontroversial and is fine with me.

**Comment by XiaocongAi** [comment]
Hi @baschlag , thank you! But I don't see your comments. Did you forget to post them?

**Comment by baschlag** [comment]
> Hi @baschlag , thank you! But I don't see your comments. Did you forget to post them?

Yeah sorry, forgot to hit the "Add review" button at the end. As I said, only two minor comments from my side.

**Comment by XiaocongAi** [comment]
Hi @baschlag , thank you! I have made the update based on your comments.

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectorySummaryWriter.hpp [hunk 1]
```diff
@@ -23,8 +23,8 @@ namespace ActsExamples {
 /// @class RootTrajectorySummaryWriter
```
[the remark this answers is not in the document]
Comment (XiaocongAi) [reply Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectorySummaryWriter.hpp:23]: Right. Fixed.

File: Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectorySummaryReader.hpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+/// @class RootTrajectorySummaryReader
+///
+/// @brief Reads in TrackParameter information from a root file
+/// and fills it into a Acts::BoundTrackParameter format
+class RootTrajectorySummaryReader : public IReader {
+ public:
+  /// @brief The nested configuration struct
+  struct Config {
+    std::string outputTracks = "outputTracks";  ///< track collection to read
+    std::string outputParticles =
```
[the remark this answers is not in the document]
Comment (XiaocongAi) [reply Examples/Io/Root/include/ActsExamples/Io/Root/RootTrajectorySummaryReader.hpp:34]: Fixed.

**Review by baschlag — APPROVED** [review]
Thanks for the updates. Looks good to me, I approve.

**Pull request merged**